### PR TITLE
add signet support

### PR DESCRIPTION
**Pull request by wakiyamap**
rebase https://github.com/spesmilo/electrum/pull/7267

Please let me know if you have any problems.

### CODE REVIEW

File: electrum/lnaddr.py [hunk 1]
```diff
@@ -178,6 +178,9 @@ def pull_tagged(stream):
     return (CHARSET[tag], stream.read(length * 5), stream)
 
 def lnencode(addr: 'LnAddr', privkey) -> str:
+    # see https://github.com/lightningnetwork/lightning-rfc/pull/844
+    if constants.net.NET_NAME == "signet":
+        addr.currency += "s"
```
Comment (SomberNight) [inline electrum/lnaddr.py:183]: I don't like this. I imagine if you call `lnencode` multiple times with the same LnAddr, it only works on the first call? As it mutates the LnAddr. I haven't tested though.
The root problem here is that `net.SEGWIT_HRP` is used for both the bech32 address prefix, and the bolt11 invoice prefix; and this duplication is not true anymore for signet. Let us simply have two constants in `net` then:
`net.SEGWIT_HRP` and `net.BOLT11_HRP`.
For most networks, they would be equal, but for signet they would be different.

Comment (wakiyamap) [reply electrum/lnaddr.py]: As far as I've tested it was fine, but it's certainly true. I fixed it.

**Comment by johnzweng** [comment]
Is there any update on Signet support? 

Now that we have Taproot locked in for activation in November, I'd like starting to play around with it also in Electrum. 🙂  
(Taproot is already enabled on Signet, but not on Testnet yet).

**Comment by SomberNight** [comment]
Thanks for the pull request.
I've pushed a follow-up commit (https://github.com/spesmilo/electrum/commit/57e52da77fd25300f6035a87c330086caf1caffc), which instead removes the `currency` field altogether.

**Comment by carnhofdaki** [comment]
Wonderful! Thanks for adding Signet! Works well.

**Pull request merged**